extensions/ql-vscode/docs/test-plan.md

```diff
@@ -0,0 +1,257 @@
+# Test Plan
+
+This document describes the manual test plan for the QL extension for Visual Studio Code.
+
+The plan will be executed manually to start with but the goal is to eventually automate parts of the process (based on 
+effort vs value basis).
+
+#### What this doesn't cover
+We don't need to test features (and permutations of features) that are covered by automated tests.
+
+#### Areas to consider
+1. Running MRVA and viewing results
+2. Running local queries
+3. Interacting with the query history
+4. Exporting MRVA results
+5. Cancelling a MRVA run
+6. Changing colour theme
+
+### Before releasing the VS Code extension
+- Check major PRs since the previous release for specific one-off things to test.
+- Run a query using the existing version of the extension (to generate an “old” query history item)
+- Go through the required test cases listed below
+
+## Required Test Cases
+
+### Test Case 1: Running a problem path query and viewing results
+
+1. Open the [UnsafeJQueryPlugin query](https://github.com/github/codeql/blob/main/javascript/ql/src/Security/CWE-079/UnsafeJQueryPlugin.ql).
+2. Run it against the following repo list:
+```
+"test-repo-list": [
+    "angular-cn/ng-nice",
+    "apache/hadoop",
+    "apache/hive"
+]
+```
+3. Check the notification message. It should:
+   - Show the number of repos that are going to be queried
+   - Provide a link to the actions workflow
+4. Check the query history. It should:
+   - Show that an item has been added to the query history
+   - The item should be marked as “in progress”.
+5. Once the query completes:
+   - Check that a notification is shown with a link to the results view 
+   - Check the results view 
+   - Check the code paths view, including the code paths drop down menu. 
+   - Check that the repository filter box works
+   - Click links to files/locations on GitHub.
+
+### Test Case 2: Running a problem query and viewing results
+
+1. Open the [ReDoS query](https://github.com/github/codeql/blob/main/javascript/ql/src/Performance/ReDoS.ql).
+2. Run it against the "Top 10" repositories.
+3. Check the notification message. It should:
+   - Show the number of repos that are going to be queried
+   - Provide a link to the actions workflow
+4. Check the query history. It should:
+   - Show that an item has been added to the query history
+   - The item should be marked as “in progress”.
+5. Once the query completes:
+   - Check that a notification is shown with a link to the results view
+   - Check that the results are rendered with an alert message and a highlighted code snippet:
+![highlighted-code-snippet](images/highlighted-code-snippet.png)
+
+### Test Case 3: Running a non-problem query and viewing results
+
+1. Open the [FunLinesOfCode query](https://github.com/github/codeql/blob/main/cpp/ql/src/Metrics/Functions/FunLinesOfCode.ql).
+2. Run it against a single repository (e.g. `google/brotli`).
```

Comment:
It's not clear to me whether this is a variant analysis against a single repository or a local query run. Could we clarify that here?